src/Std/Time/Zoned/ZonedDateTime.lean

```diff
@@ -14,7 +14,8 @@ namespace Time
 -- set_option linter.all true
 
 /--
-Represents a date and time with timezone information.
+Represents a date and time with timezone information. Boolean equality (`BEq`/`==`) of two instances
+means that the underlying time zone independent timestamps are equal.
```

Comment:
I don't think this is a good choice. It violates what a user may reasonably expect the equality to mean and it also differs from other programming languages. For example, Java compares like this: "The comparison is based first on the instant, then on the local date-time, then on the zone ID, then on the chronology."

I think this is a bit subtle, so I'd be happy to just not provide the instance in this PR and discuss this in more detail later.